### ISSUE TITLE
Amazon echo doesn't discover hue :( But Philips app does!

### ISSUE THREAD

**Issue by Echotrouble** [bwssytems/ha-bridge#350]
Hello everybody,
I just set up bws HA bridge on a raspberry pi. Now I want to connect it to my Amazon Echo, but it is not being discovered. The strange thing is that on the Phillips hue app, I can find the HA bridge and I am able to use it as if it was a real Philips bridge.

I should also note two other things: There are two Phillips apps, one wants to update the hue bridge, the other one doesn't care and I can use the 'Philips hue bridge'. My Logitech Harmony also doesn't discover the HA bridge :(
All the devices are in the same network.

What could be the problem and how do I investigate?

My web server is running at port 80 and the UPNP response port is running at port 50000.
Here is some netstat output:

Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name

tcp6       0      0 :::80                   :::*                    LISTEN      -
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           -
udp        0      0 0.0.0.0:59208           0.0.0.0:*                           -
udp6       0      0 :::5353                 :::*                                -
udp6       0      0 :::43548                :::*                                -
udp6       0      0 :::50000                :::*                                -
udp6       0      0 :::1900                 :::*                                -

**Comment by audiofreak9** [comment]
Make sure your Echo and RPi are on the same wireless frequency (if applicable), typically 2.4GHz.  <a href="http://www.techtimes.com/articles/125257/20160118/amazon-echo-issues-and-quick-fixes-wi-fi-connectivity-connected-home-device-how-to-reset-and-more.htm">Article here for reference</a>.

Change the ha-bridge config option "Emulate Hue Hub Version" to 01036659 so the update required stops.  Just know you can NOT create a room on the official HUE gen 2 app.

You said _"Logitech Harmony also doesn't discover the HA bridge"_, just want to make sure you mean your Harmony Hub?  The Harmony Hub appears as if it is not connecting.  I had to use the manual add option and wait out the "Press the button on the bridge" prompt.

The next step would be to turn on UPNP tracing in the logs.

**Comment by Echotrouble** [comment]
Okay thank you.
The Pi is connected to the wired network, echo to a wifi hotspot. I already enabled upnp tracing and saw the requests coming in from the  Philips app. But nothing form the echo.

I did do a wireshark and saw the amazon echo upnp requests (on my computer).


And indeed I was talking about logitech hub.

**Comment by audiofreak9** [comment]
Wired and wireless are the same router/modem?

Did you reinitialize the ha-bridge?

Have you rebooted the router, RPi and the Echo?

**Comment by Echotrouble** [comment]
Yeah, everything is in the same subnet. Nothing is being blocked as far as I can see (router is a brand new TP-Link AC router). Already rebooted Pi, router and echo :/

**Comment by audiofreak9** [comment]
Router has UPNP turned on of course?...

I had a few other people I've [helped that only had success when they reinstalled Raspbian](http://coreyswrite.com/tips-tricks/amazon-echo-x10-home-control-updated/comment-page-6/#comment-1743), then the ha-bridge.  It sucks to have to do all that work, but you can save your device.db file and restore it once ha-bridge is installed.

**Comment by Echotrouble** [comment]
Hmm okay, will further investigate. 

How configure the "Logitech Harmony also doesn't discover the HA bridge" ?

**Comment by Echotrouble** [comment]
Holy shit! My alexa just discovered the bridge. What I did differently this time?
I ran tcp dump at the same time I asked alexa to discover devices. Very very weird. But hey it works now, I'm not complaining!

**Comment by bwssytems** [comment]
This would mean there is an issue with your OS install or network settings. tcpdump puts the network interface stack into promiscuous mode and changes its behavior. You may want to do what @audiofreak9 suggested and reinstall raspbian.

**Comment by peterosburg** [comment]
I actually had the same issue. Running latest raspbian with an apache on port 80 and the habridge on 8080. I couldn't get Alexa to discover anything, so I changed the ports and the bridge is now running on port 80. It immediately solved the issue and my amazon echo was detecting the devices.

**Comment by Echotrouble** [comment]
But my bridge is already running at port 80.

**Comment by bwssytems** [comment]
This issue is not a bridge issue, is this resolved?

**Comment by Echotrouble** [comment]
I haven't find the time to reinstall the pi. however I don't think it will
change anything. this was a brand new install and th eproblem is quite
consistent. The same for my Logitech harmony hub discovery. I had to run
tcpdump again in order for it to discoverthe emulator. Can this be caused
by the cheap chinese usb to ethernet I'm using?

**Comment by bwssytems** [comment]
So, you are using a usb to wifi adapter? That is most likely your cause. You will need to update or do some advanced settings on your driver as it sounds like it is going to sleep. Can you attach to the router as wired? Would be better.

**Comment by Echotrouble** [comment]
Hello,

no I'm using a cheap USB to ethernet adapter:
http://www.banggood.com/Micro-USB-2_0-to-Ethernet-10-or-100-RJ45-Network-Lan-Adapter-Card-p-921585.html?rmmds=search
But I've ordered a USB adapter (micro usb to normal) so I can test with
other 'usb to ethernet adapters' and 'WiFi adapters' as well, I'll keep you
updated (however it will take some time until I receive the USB adapter
(micro usb to normal).

**Comment by bwssytems** [comment]
So, are your using a Pi Zero or an older Pi 1 Model A? Those are the ones without wired ethernet ports.

**Comment by Echotrouble** [comment]
I'm using a raspberry pi zero. With the white usb to ethernet adapter. I
used PiBakery to create the flash card with ssh login.

**Comment by bwssytems** [comment]
So, this is something you will need to research on the Raspberry Pi sites as the bridge is not your issue. This is not an issue list for system problems.

**Comment by Zippochonda** [comment]
Thanks for this Thread, i also tried to install it on a raspberry zero with the white adapter. It didnt worked, now the same configuration is running on a raspberry 3. Without this disscussion i would never ever think on the ethernetadapter...

**Comment by Echotrouble** [comment]
Aah cool I am not alone.

Did you try a 'tcpdump -c 2000' or something similar while running the discovery? And did it work then?

**Comment by trcarlso** [comment]
Where do I need to do this "Change the ha-bridge config option "Emulate Hue Hub Version" to 01036659 so the update required stops." I don't see an option in the webgui, is this something I need to do in a file on my raspberry pi?

**Comment by audiofreak9** [comment]
This setting is only available in version 4.0, which is in beta testing currently.  You can follow it and download to tweet it here in issue https://github.com/bwssytems/ha-bridge/issues/355

**Comment by bwssytems** [comment]
Is this resolved?

**Comment by Echotrouble** [comment]
Hello, I'm still waiting for the small adapter piece to place between my raspberry pi zero and my usb wifi dongle. When I got that adapter I can determine if it is a hardware or software problem.

**Comment by bwssytems** [comment]
Again, This is an issue with your system and not the ha-bridge.... This has been identified as an issue with the dongle you are using.

**Comment by InRong** [comment]
Strangely enough, due to these issues, I deleted my devices, but it would not rediscover them. 

I tried v5.3.0 and v5.2.1, neither of them worked. However I reverted to version 4.2.1 as I had it on the pi, and all my devices are back again, and working.... I had been running v5.2.1 when this all started, but grabbed v5.3.0 to try to solve it.

Is there something different in the discovery response between v4 and v5?

Just checked, and for me v5.3.0 does not work, I will have to remain reverted to v4.2.1. This is running on port 80. Before the updates v5.2.1 was working fine although this was on port 8080. I made the switch as I needed to move to a newer pi, and because of the details listed in here.

I have echo dots, one v2 and one v3, so I am not sure which one is being master when I use the app to discover.
